### ISSUE TITLE
HttpServletRequest does not handle Huge Content-Length properly

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#3503]
If a large request arrives, such as ...

```
POST / HTTP/1.1
Host: whatever
Content-Type: application/octet-stream
Content-Length: 21474836470
Connection: close

<insert huge amount of content here>
```

The request header for `Content-Length` is a value too large for an Integer.

The javadoc for `HttpServletRequest.getContentLength()` says that this should return a value of `-1`.
See: https://docs.oracle.com/javaee/7/api/javax/servlet/ServletRequest.html#getContentLength--

> **Returns:**
> an integer containing the length of the request body or -1 if the length is not known or is greater than Integer.MAX_VALUE.

The servlet api user could then use the `getContentLengthLong()` to obtain the real value.

**Comment by joakime** [comment]
I pushed a commit with 2 testcases in branch `jetty-9.4.x-3503-huge-content-length`.

While the test that exceeds MAX_INTEGER actually reaches the Servlet, the behaviors of `getContentLength()` isn't valid.

The other test, that exceeds MAX_LONG is failed with an error 400.
```
HTTP/1.1 400 Invalid Content-Length Value
Content-Type: text/html;charset=iso-8859-1
Content-Length: 71
Connection: close
Server: Jetty(9.4.z-SNAPSHOT)

<h1>Bad Message 400</h1><pre>reason: Invalid Content-Length Value</pre>
```

But this isn't following the specs either.
See https://tools.ietf.org/html/rfc7230#page-31 (section about `Content-Length`)

>    Any Content-Length field value greater than or equal to zero is
>    valid.  Since there is no predefined limit to the length of a
>    payload, a recipient MUST anticipate potentially large decimal
>    numerals and prevent parsing errors due to integer conversion
>    overflows (Section 9.3).

[Section 9.3 - Attacks via Protocol Element Length](https://tools.ietf.org/html/rfc7230#section-9.3) points out the following ...

>    A server can reject a message that has a request-target that is too
>    long (Section 6.5.12 of [RFC7231]) or a request payload that is too
>    large (Section 6.5.11 of [RFC7231]).  Additional status codes related
>    to capacity limits have been defined by extensions to HTTP [RFC6585].

Which links to https://tools.ietf.org/html/rfc7231#section-6.5.11 - HTTP Status code 413 Payload Too Large

The use case of exceeding MAX_LONG should return 413, not 400.

**Comment by joakime** [comment]
This has been fixed as part of PR #3889
